plugin/evm/acp176/acp176_test.go

```diff
@@ -0,0 +1,694 @@
+			expected: State{
+				Gas: gas.State{
+					Capacity: 890_000,
+					Excess:   2_110_000,
+				},
+			},
+		},
+		{
+			name: "gas_used_capacity_exceeded",
+			initial: State{
+				Gas: gas.State{
+					Capacity: 1_000_000,
+					Excess:   2_000_000,
+				},
+			},
+			gasUsed:      1_000_001,
+			extraGasUsed: nil,
+			expectedErr:  gas.ErrInsufficientCapacity,
+			expected:     State{},
+		},
+		{
+			name: "massive_extra_gas_used",
+			initial: State{
+				Gas: gas.State{
+					Capacity: 1_000_000,
+					Excess:   2_000_000,
+				},
+			},
+			gasUsed:      0,
+			extraGasUsed: new(big.Int).Lsh(common.Big1, 64),
+			expectedErr:  gas.ErrInsufficientCapacity,
+			expected: State{
+				Gas: gas.State{
+					Capacity: 1_000_000,
+					Excess:   2_000_000,
+				},
+			},
+		},
+		{
+			name: "extra_gas_used_capacity_exceeded",
+			initial: State{
+				Gas: gas.State{
+					Capacity: 1_000_000,
+					Excess:   2_000_000,
+				},
+			},
+			gasUsed:      0,
+			extraGasUsed: big.NewInt(1_000_001),
+			expectedErr:  gas.ErrInsufficientCapacity,
+			expected:     State{},
+		},
+	}
+	updateTargetExcessTests = []struct {
+		name                string
+		initial             State
+		desiredTargetExcess gas.Gas
+		expected            State
+	}{
+		{
+			name: "no_change",
+			initial: State{
+				Gas: gas.State{
+					Excess: 2_000_000,
+				},
+				TargetExcess: 0,
+			},
+			desiredTargetExcess: 0,
+			expected: State{
+				Gas: gas.State{
+					Excess: 2_000_000,
+				},
+				TargetExcess: 0,
+			},
+		},
+		{
+			name: "max_increase",
+			initial: State{
+				Gas: gas.State{
+					Excess: 2_000_000,
+				},
+				TargetExcess: 0,
+			},
+			desiredTargetExcess: MaxTargetExcessDiff + 1,
+			expected: State{
+				Gas: gas.State{
+					Excess: 2_001_954, // 2M * NewTarget / OldTarget
+				},
+				TargetExcess: MaxTargetExcessDiff, // capped
+			},
+		},
+		{
+			name: "inverse_max_increase",
+			initial: State{
+				Gas: gas.State{
+					Excess: 2_001_954,
+				},
+				TargetExcess: MaxTargetExcessDiff,
+			},
+			desiredTargetExcess: 0,
+			expected: State{
+				Gas: gas.State{
+					Excess: 2_000_000, // inverse of max_increase
+				},
+				TargetExcess: 0,
+			},
+		},
+		{
+			name: "max_decrease",
+			initial: State{
+				Gas: gas.State{
+					Excess: 2_000_000_000,
+				},
+				TargetExcess: 2 * MaxTargetExcessDiff,
+			},
+			desiredTargetExcess: 0,
+			expected: State{
+				Gas: gas.State{
+					Excess: 1_998_047_816, // 2M * NewTarget / OldTarget
+				},
+				TargetExcess: MaxTargetExcessDiff,
+			},
+		},
+		{
+			name: "inverse_max_decrease",
+			initial: State{
+				Gas: gas.State{
+					Excess: 1_998_047_816,
+				},
+				TargetExcess: MaxTargetExcessDiff,
+			},
+			desiredTargetExcess: 2 * MaxTargetExcessDiff,
+			expected: State{
+				Gas: gas.State{
+					Excess: 1_999_999_999, // inverse of max_decrease -1 due to rounding error
+				},
+				TargetExcess: 2 * MaxTargetExcessDiff,
+			},
+		},
+	}
+)
+
+func TestTarget(t *testing.T) {
```

Comment:
`TestTarget`, `TestMaxCapacity`, `TestBaseFee` and `TestDesiredTargetExcess` could be merged into one test I think. I will also say sharing a global variable of tests is a risky approach in case one test modifies one test case (now or in the future) which would influence other tests.

Reply:
I don't have an opinion here. What would this test look like?
```golang
func TestEvaluate(t *testing.T) {
	for _, test := range readerTests {
		t.Run(test.name, func(t *testing.T) {
			assert := assert.New(t)

			assert.Equal(test.target, test.state.Target(), "incorrect target")
			assert.Equal(test.maxCapacity, test.state.MaxCapacity(), "incorrect maxCapacity")
			assert.Equal(test.gasPrice, test.state.GasPrice(), "incorrect gasPrice")
			if !test.skipTestDesiredTargetExcess {
				assert.Equal(test.state.TargetExcess, DesiredTargetExcess(test.target), "incorrect desiredTargetExcess")
			}
		})
	}
}
```
?